microservices/case-service/src/main/java/org/entando/bundle/controller/CaseController.java

```diff
@@ -1,71 +1,134 @@
 
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 
+import static org.entando.bundle.BundleConstants.CASE_MANAGEMENT_ADMIN;
+
 @RestController
-@RequestMapping("/api")
+@RequestMapping("/api/cases")
 @CrossOrigin(origins = "*", allowedHeaders = "*")
 public class CaseController {
 
-    private final CaseService processService;
+    private Logger log = LoggerFactory.getLogger(CaseController.class);
 
-    private final FileService fileService;
+    private final CaseService caseService;
 
-    public CaseController(CaseServiceImpl processService, FileServiceImpl fileService) {
-        this.processService = processService;
-        this.fileService = fileService;
+    public CaseController(CaseServiceImpl caseService) {
+        this.caseService = caseService;
     }
 
-    @GetMapping("/processes")
-    @PreAuthorize("hasAnyAuthority('case-management-admin')")
-    public @ResponseBody List<Process> getProcesses() {
-        return processService.getAllProcesses();
+    /**
+     * Check whether the current user is also administrator of cases
+     * @param authorities the authorities of the currently logged user
+     * @return true if the current user has administration rights
+     */
+    private boolean isRoleAdmin(Collection<? extends GrantedAuthority> authorities) {
+        Optional<? extends GrantedAuthority> auth = authorities.stream().filter(a -> a.getAuthority().equals(CASE_MANAGEMENT_ADMIN)).findFirst();
+        return auth.isPresent();
     }
 
-    @GetMapping("/process/{id}")
-    @PreAuthorize("hasAnyAuthority('case-management-admin')")
-    public @ResponseBody Optional<Process> getProcess(@PathVariable Long id) {
-        Optional<Process> process = processService.getProcess(id);
-        return process;
+    @GetMapping
+    @PreAuthorize("hasAnyAuthority('case-management-admin','case-management-entry')")
+    public @ResponseBody List<Case> getCases(Authentication authentication, Principal principal) {
+        try {
+            log.info("REST to get the case list");
+            if (isRoleAdmin(authentication.getAuthorities())) {
+                return caseService.getAllCases();
+            } else {
+                log.debug("returning Case list for the user {}", principal.getName());
+                return caseService.getCasesByName(principal.getName());
+            }
+        } catch (Throwable t) {
+            log.error("error getting the list of cases", t);
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Unexpected getting the Cases list", t);
+        }
     }
 
-    @PostMapping("/processes")
+    @GetMapping("/{id}")
+    @PreAuthorize("hasAnyAuthority('case-management-admin','case-management-entry')")
+    public @ResponseBody Object getCase(Authentication authentication, Principal principal, @PathVariable Long id) {
```

Comment:
Object -> Optional < Case >

Reply:
I kept things simple - I need the generic object to be able to return the custom 404 payload

Reply:
Done